File: Competitive Programming 3/Section 1.3.3/10550 - Combination Lock/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

int difPositions(int orig, int dest, bool clockwise){
    if(clockwise){
        swap(orig,dest);
    }
    return dest<=orig ? orig - dest : 40 - dest + orig ; // ternary expression
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int init, a, b, c;
    while(cin>>init>>a>>b>>c,init||a||b||c){
        cout << (difPositions(init,a,false) + difPositions(a,b,true) + difPositions(b,c,false)) *9 + 1080 << "\n";
    }
    return 0;
}

File: Competitive Programming 3/Section 1.3.3/10550 - Combination Lock/Java/Main.java
import java.util.Scanner;

public class Main {

    static int difPositions(int orig,int dest, boolean clockwise){
        if(clockwise){
            //swap orig = A , dest = B
            orig^=dest;  // A ->B, B -> A
            dest^=orig;
            orig^=dest;
        }
        return dest < orig ? orig-dest : 40 - dest + orig;
    }

    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        while(true){
            int init = s.nextInt(), a = s.nextInt(), b = s.nextInt(), c = s.nextInt();
            if(init==0&&a==0&&b==0&&c==0){
                break;
            }
            System.out.printf("%d\n",(difPositions(init,a,false) + difPositions(a,b,true) + difPositions(b,c,false)) * 9 + 1080);
        }
    }
}